packages/web/app/src/server/slack.ts

```diff
@@ -130,9 +182,65 @@ export function connectSlack(server: FastifyInstance) {
     const { organizationSlug } = paramsResult.data;
     req.log.info('Connect organization to Slack (id=%s)', organizationSlug);
 
-    const slackUrl = `https://slack.com/oauth/v2/authorize?scope=incoming-webhook,chat:write,chat:write.public,commands&client_id=${env.slack.clientId}`;
-    const redirectUrl = `${env.appBaseUrl}/api/slack/callback`;
-
-    void res.redirect(`${slackUrl}&state=${organizationSlug}&redirect_uri=${redirectUrl}`);
+    void res.redirect(
+      SlackAPI.createOauth2AuthorizeUrl({
+        clientId: env.slack.clientId,
+        redirectUrl: `${env.appBaseUrl}/api/slack/callback`,
+        scopes: ['incoming-webhook', 'chat:write', 'chat:write.public', 'commands'],
+        state: organizationSlug,
+      }),
+    );
   });
 }
+
+// =================================
+// Error Classes
+// =================================
+
+// eslint-disable-next-line @typescript-eslint/no-namespace
+namespace SlackIntegrationErrors {
+  export class DisabledError extends Kit.Errors.ContextualError<'DisabledError'> {
+    message = 'The Slack integration is not enabled.';
```

Comment:
Are there any recommendations you could provide in this message for how to correct this if it needs correcting?
e.g. "set environment variable X" or "see <url>"

Reply:
We receive a string from the Slack API, quite lame. I have not attempted to increase the quality here by parsing it, reviewing the Slack docs, if any exist for this string, etc. It would definitely be an improvement! Worth at least a TODO here I agree.

Reply:
Definitely worth putting a TODO about that here. We could write a catch all documentation here or link to some docs. Since this case is classified, however, upstream error rendering has the potential to switch on this and provide recommendations where it by definition has more context.

Error rendering is a really important thing that isn't addressed in this PR. For example that is where localization would be factored in, I think. Keeping strings out of error classes is generally good. But as you've pointed out in a few places, without good rendering, all that context and structure is just potential energy, not realized.